README.md

```diff
@@ -4,142 +4,232 @@ Containers to be used for general purpose Data Science.
+**Recommendation:** start from a small official base image and build layers
+ontop of it.
 
 ### Scan your containers for vulnerabilities
 
-- There are lots of tools to do this in an automated way (e.g. [trivy](https://github.com/aquasecurity/trivy))
+- There are lots of tools to do this in an automated way (e.g. [trivy][])
 
-- Can build a docker image and scan it as part of a continuous integration pipeline. E.g. a condition to merge to master is that the Dockerfile provided must build successfully and then pass a security scan.
+- Can build a docker image and scan it as part of a continuous integration
+  pipeline. E.g. a condition to merge to master is that the Dockerfile provided
+  must build successfully and then pass a security scan.
 
-**Recommendation:** scan images for vulnerabilities, and consider incorporating this process as a job in your CI/CD pipeline.
+**Recommendation:** scan images for vulnerabilities, and consider incorporating
+this process as a job in your CI/CD pipeline.
 
-### Stanzas should be ordered from least likely to change to most likely to change
+### Order stanzas from least likely to change to most likely to change
 
 - Each line in your `Dockerfile` is called a stanza.
 
-- When you are building an image, Docker is creating and caching an image layer for each stanza in your `Dockerfile`. Since layers are cached, if you build your image multiple times (e.g. you are changing certain stanzas), Docker can reuse the cached layers to reduce build time.
+- When you are building an image, Docker is creating and caching an image layer
+  for each stanza in your `Dockerfile`. Since layers are cached, if you build
+  your image multiple times (e.g. you are changing certain stanzas), Docker can
+  reuse the cached layers to reduce build time.
 
-- However, any stanzas that come after the changed stanza **will be rebuilt** and therefore do not benefit from this layer caching that Docker implements.
+- However, any stanzas that come after the changed stanza **will be rebuilt**
+  and therefore do not benefit from this layer caching that Docker implements.
 
-**Recommendation:** to take advantage of layer caching and reduce your image build times, it is recommended to put expensive stanzas that don't change often early in your Dockerfile, and put lighter stanzas that do change often at the end. This allows you to avoid rebuilding expensive unchanging layers every time you need to rebuild your image after making a small change (e.g. installing a new Python package).
+**Recommendation:** to take advantage of layer caching and reduce your image
+build times, it is recommended to put expensive stanzas that don't change often
+early in your Dockerfile, and put lighter stanzas that do change often at the
+end. This allows you to avoid rebuilding expensive unchanging layers every time
+you need to rebuild your image after making a small change (e.g. installing a
+new Python package).
 
```

Comment:
```suggestion

### Handling python packages in Dockerfiles

- In general it is good practice to use a `requirements.txt` file in a project and to pin the versions (`pandas==1.2.3` rather than just `pandas`).  Requirements files are also useful in constructing more succinct Dockerfiles (`RUN pip install -r requirements.txt` is briefer than `RUN pip install package1 package2 ...`)

- But, sometimes `requirements.txt` files can cause issues in Dockerfiles.  For example, if you have 10 packages that never change but one that is updated frequently, you have the same problem as when you put frequently changed items at the top of a Dockerfile.  Sometimes it is appropriate to have `requirements_rarely_changing.txt` and `requirements_frequently_changing.txt` (or, perhaps better titles), this way the frequently changing packages can be installed later in the Dockerfile.  

```